src/parsers/perf/perfparser.cpp

```diff
@@ -988,6 +988,22 @@ class PerfParserPrivate : public QObject
 
     void addSample(const Sample& sample)
     {
+        if (m_lastTime != -1) {
+            for (const auto& cost : sample.costs) {
+                if (frequencyResult.data.size() <= cost.attributeId) {
+                    frequencyResult.data.resize(cost.attributeId + 1);
+                    frequencyResult.name.resize(cost.attributeId + 1);
+                    frequencyResult.name[cost.attributeId] = strings.at(attributes[cost.attributeId].name.id);
+                }
+
+                const Data::FrequencyData frequencyData = {
+                    sample.time, static_cast<double>(cost.cost) / ((sample.time - m_lastTime))};
+                frequencyResult.data[cost.attributeId].push_back(frequencyData);
+            }
+        }
+
+        m_lastTime = sample.time;
```

Comment:
once the time is tracked per core, this should then be something like this instead (imo value 0 for an invalid time is fine):

```
auto &lastTime = m_lastSampleTimePerCore[sample.cpu];
if (lastTime) {

}

lastTime = sample.time;
```